app/models/order.rb

```diff
@@ -1,21 +1,25 @@
 class Order < ApplicationRecord
+  include TokenGenerator
 
   ## ASSOCIATIONS
   belongs_to :user
   has_many :line_items, dependent: :destroy
   belongs_to :address, optional: true
   has_many :deals, through: :line_items
+  has_one :payment
 
   ## SCOPES
   scope :incomplete, ->{ where(completed_at: nil) }
   scope :complete, ->{ where.not(completed_at: nil) }
 
   ## CALLBACKS
   before_destroy :ensure_order_incomplete
+  after_commit :generate_invoice_number, on: :create
 
   ## STATE MACHINE
   state_machine :state, initial: :cart do
```

Comment:
extract state_machine logic and all method related to state_machine in checkout.rb